tidy3d/components/base.py

```diff
@@ -563,7 +574,12 @@ def to_hdf5(self, fname: str, custom_encoders: List[Callable] = None) -> None:
 
         with h5py.File(fname, "w") as f_handle:
 
-            f_handle[JSON_TAG] = self._json_string
+            json_str = self._json_string
+            for ind in range(ceil(len(json_str) / MAX_STRING_LENGTH)):
+                key = f"{JSON_TAG}_{ind}" if ind > 0 else JSON_TAG
```

Comment:
Maybe we can write a static method or module level function for this to avoid duplicating the key as a function of index, incase we change this or need to use it elsewhere.

Reply:
in Tidy3DBaseModel
```py
@staticmethod
def _json_string_key(index):
    if index:
        return f"{JSON_TAG}_{ind}"
    return JSON_TAG
```

`to_hdf5`
```py
json_str = self._json_string
num_strings = ceil(len(json_str) / MAX_STRING_LENGTH)
for ind in range(num_strings):
    ind_start = int(ind * MAX_STRING_LENGTH)
    ind_stop = min(int(ind + 1) * MAX_STRING_LENGTH, len(json_str))
    key = self. _json_string_key(ind)
    f_handle[key] = json_str[ind_start:ind_stop]
```

```py
@classmethod
def _json_string_from_hdf5(cls, fname: str) -> str:
         """Load the model json string from an hdf5 file."""
         with h5py.File(fname, "r") as f_handle:
               num_string_parts = len([key for key in f_handle.keys() if JSON_TAG in key])
               json_string = ""
               for ind in range(num_string_parts):
                     key = _json_string_key(ind)
                     json_string += f_handle[key][()]
           return json_string
```